rclcpp_action/src/client_base.cpp

```diff
@@ -386,7 +386,7 @@ ClientBase::handle_goal_response(
   const rmw_request_id_t & response_header,
   std::shared_ptr<void> response)
 {
-  std::lock_guard<std::mutex> guard(pimpl_->goal_requests_mutex);
+  std::lock_guard<std::recursive_mutex> guard(pimpl_->goal_requests_mutex);
```

Comment:
this allows us to request another goal to the action server on previous goal response (accept or reject) is arrived.